src/cdk-experimental/scrolling/virtual-scroll.md

```diff
@@ -0,0 +1,143 @@
+**Warning: this component is still experimental. It may have bugs and the API may change at any
+time**
+
+`<cdk-virtual-scroll-viewport>` is used to display a scrolling list containing a large number of
+items that would have a negative effect on performance if all rendered at once. Instead the virtual
+scroll viewport renders each item as needed when the user scrolls it into view. The
+`<cdk-virtual-scroll-viewport>` works together with the `*cdkVirtualFor` directive which is used to
+render items inside of the viewport. It also requires that a `VirtualScrollStrategy` be provided.
+The easiest way to provide a strategy is with one of the directives `itemSize` or `autosize`.
+
+<!-- TODO(mmalerba): Add embedded examples once this is available in @angular/cdk -->
+For some example usages, 
+[see the demo app](https://github.com/angular/material2/tree/master/src/demo-app/virtual-scroll).
+
+### Creating items in the viewport
+`*cdkVirtualFor` should be used instead of `*ngFor` to create items in the
+`<cdk-virtual-scroll-viewport>`. `*cdkVirtualFor` supports all of the same features as `*ngFor`.
+The simplest usage just specifies the list of items:
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  <div *cdkVirtualFor="let item of items">{{item}}</div>
+</cdk-virtual-scroll-viewport>
+```
+
+`*cdkVirtualFor` makes the following context variables available to the template:
+* `index` - The index of the item in the data source.
+* `count` - The total number of items in the data source.
+* `first` - Whether this is the first item in the data source.
+* `last` - Whether this is the last item in the data source.
+* `even` - Whether the `index` is even.
+* `odd` - Whether the `index` is odd.
+
+All of these apply to the index of the item in the data source, not the index in the rendered
+portion of the data.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  <div *cdkVirtualFor="let item of items;
+                       let index = index;
+                       let count = count;
+                       let first = first;
+                       let last = last;
+                       let even = even;
+                       let odd = odd">
+    {{item}} - {{index}} - {{count}} - {{first}} - {{last}} - {{even}} - {{odd}}
+  </div>
+</cdk-virtual-scroll-viewport>
+```
+
+A `trackBy` function can be specified and works the same as the `*ngFor` `trackBy`. The `index`
+passed to the tracking function will be the index in the data source, not the index in the rendered
+portion.
+
+#### View recycling
+In order to improve performance `*cdkVirtualFor` saves the views it creates in a cache when they are
```

Comment:
Editing for brevity and passive voice:
```markdown
To improve rendering performance, `*cdkVirtualFor` caches previously created views after
they are no longer needed. When a new view would normally be created, a cached view
is reused instead. The size of the view cache can be adjusted via the `templateCacheSize`
property; setting this size to `0` disables caching.
```

It would also be good to explain _why_ you might want to change the cache size.